tests/wp-unit/include/Core/Metas/Modules/DeleteUserMetaModuleTest.php

```diff
@@ -152,5 +182,294 @@ public function test_delete_usermeta_in_batches() {
+	 */
+	public function test_that_delete_multiple_users_metas_fields_from_multiple_user_role_no_users_and_have_users() {
+
+		$role_array_without_users = array();
+		for ( $i = 0; $i < 10; $i++ ) {
+			$role                       = 'user_type_no_user_' . $i;
+			$role_array_without_users[] = $role;
+		}
+
+		$role_array_with_users = array();
+		for ( $i = 0; $i < 10; $i++ ) {
+			$role                    = 'user_type_' . $i;
+			$role_array_with_users[] = $role;
+			// Create a users with meta value in dynamic role.
+			for ( $j = 0; $j < 10; $j++ ) {
+				$user = $this->factory->user->create( array( 'role' => $role ) );
+				add_user_meta( $user, 'time', '10/10/2018' );
+			}
+		}
+
+		$user_another = $this->factory->user->create( array( 'role' => 'administrator' ) );
+
+		// call our method.
+		$delete_options = array(
+			'selected_roles' => $role_array_without_users,
+			'meta_key'       => 'time',
+			'use_value'      => false,
+			'limit_to'       => 0,
+			'delete_options' => '',
+		);
+
+		$meta_deleted = $this->module->delete( $delete_options );
+		$this->assertEquals( 0, $meta_deleted );
+
+		// call our method.
+		$delete_options = array(
+			'selected_roles' => $role_array_with_users,
+			'meta_key'       => 'time',
+			'use_value'      => false,
+			'limit_to'       => 0,
+			'delete_options' => '',
+		);
+
+		$meta_deleted = $this->module->delete( $delete_options );
+		$this->assertEquals( 100, $meta_deleted );
+
+		$exist_user = $this->is_user_id_exists( $user_another );
+
+		$this->assertEquals( true, $exist_user );
+
+	}
+
+	/**
+	 * Test deletion of user metas with both meta key and value.
+	 */
+	public function test_that_delete_user_meta_fields_both_key_and_value() {
+		// Create a user with subscriber role.
+		$user = $this->factory->user->create( array( 'role' => 'subscriber' ) );
+
+		add_user_meta( $user, 'test_key', 'Test Value' );
+
+		$user_another = $this->factory->user->create( array( 'role' => 'subscriber' ) );
+
+		// call our method.
+		$delete_options = array(
+			'selected_roles' => array( 'subscriber' ),
+			'meta_key'       => 'test_key',
+			'meta_value'     => 'Test Value',
+			'use_value'      => true,
+			'limit_to'       => 0,
+			'delete_options' => '',
+		);
+
+		$meta_deleted = $this->module->delete( $delete_options );
+		$this->assertEquals( 1, $meta_deleted );
+
+		$user_meta = get_user_meta( $user, 'test_key', true );
+		$this->assertEquals( '', $user_meta );
+
+		$exist_user = $this->is_user_id_exists( $user_another );
+
+		$this->assertEquals( true, $exist_user );
+	}
+
+	/**
+	 * Check is user id is exist.
+	 *
+	 * @param int $user_id User id.
+	 */
+	public function is_user_id_exists( $user_id ) {
+
+		global $wpdb;
+
+		$count = $wpdb->get_var( $wpdb->prepare( "SELECT COUNT(*) FROM $wpdb->users WHERE ID = %d", $user_id ) );
+
+		if ( 1 == $count ) {
```

Comment:
Replace it with just `return ($count > 0)`

Reply:
Fixed